### PR TITLE
[BE][Ez]: Fix docs recommending inefficient tensor op order

### PR DESCRIPTION
**Pull request by Skylion007**
`detach().clone()` is faster than `.clone().detatch()` since the gradients are not cloned. Let's update all the documentation and tests so that users do not use the inefficient op ordering.

### CODE REVIEW

**Review by awgu — APPROVED** [review]
This looks reasonable to me!

**Review by XuehaiPan — APPROVED** [review]

**Comment by XuehaiPan** [comment]
Maybe we can add similar changes to `.cpu().detach().numpy()`.
